src/cmd/build.js

```diff
@@ -103,15 +111,49 @@ async function defaultPackageCreator({
   let packageName = safeFileName(
     `${extensionName}-${manifestData.version}.zip`);
   let extensionPath = path.join(artifactsDir, packageName);
-  let stream = createWriteStream(extensionPath);
 
-  stream.write(buffer, () => stream.end());
+  let destinationFileExists = await fs.stat(extensionPath)
+    .then((stat)=>{
+      return stat.isFile();
+    })
+    .catch(()=>{
```

Comment:
It's not safe to assume that all errors mean that the destination file doesn't exist. For example, you could receive an error saying that the Node program doesn't have permissions to read the destination file. This error should not be ignored, it should be thrown. 

To resolve this you can use a pattern like [isErrorWithCode()](https://github.com/mozilla/web-ext/blob/b050f119ebda01c89c33fd8572e6fd6d40ee7adc/src/util/artifacts.js#L19). I'm not sure which code means 'does not exist' but the easiest way to find out is to write a unit test that sets up a non-existant file path and see what code you get.